pkg/controller/release/strategy_executor.go

```diff
@@ -147,7 +149,30 @@ func (e *StrategyExecutor) Execute(prev, curr, succ *releaseInfo) (bool, []Strat
 	pipeline := NewPipeline()
 	pipeline.Enqueue(genInstallationEnforcer(ctx, curr, succ))
 
-	if isHead {
+	// when a release is steppingBackwards, the strategy executor needs to reverse it's direction.
+	// a release is "not steppingBackwards" when the targetStep >= achieved step.
+	// we include the state where the steps are equal because this state does not require
+	// reversing the direction of the strategy executor
+	if e.steppingBackwards {
+		// release is steppingBackwards (achieved step > target step):
+		// increase capacity for previous release
+		// increase traffic for previous release
+		// reduce traffic for current release
+		// reduce capacity for current release
+		if isHead {
+			if hasTail {
+				prevctx := ctx.Copy()
+				prevctx.isHead = false
+				pipeline.Enqueue(genCapacityEnforcer(prevctx, prev, curr))
+				pipeline.Enqueue(genTrafficEnforcer(prevctx, prev, curr))
+			}
+			pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
+			pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
+		} else {
+			pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
+			pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
+		}
```

Comment:
It seems that the block:
```go
			pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
			pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
```
is the same in both branches of the `if` statement, could we have it just once?

E.g.

```go
		if isHead && hasTail {
			prevctx := ctx.Copy()
			prevctx.isHead = false
			pipeline.Enqueue(genCapacityEnforcer(prevctx, prev, curr))
			pipeline.Enqueue(genTrafficEnforcer(prevctx, prev, curr))
		}
		pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
		pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
```